spec/triggers/aws-cloudwatch.md

```diff
@@ -0,0 +1,137 @@
+# AWS Cloudwatch Trigger
+
+This specification describes the `aws-cloudwatch` trigger for AWS Cloudwatch.
+
+```yaml
+triggers:
+  - type: aws-cloudwatch
+    metadata:
+      # Required: namespace
+      namespace: AWS/SQS
+      # Required: Dimension Name
+      dimensionName: QueueName
+      dimensionValue: keda
+      metricName: ApproximateNumberOfMessagesVisible
+      targetMetricValue: "2"
+      minMetricValue: "0"
+      # Required: region
+      awsRegion: "eu-west-1"    
+```
+
+The `namespace` value is the service you want to get metrics for, in this case 'AWS/SQS'. Many other options exist.
+
+The `dimensionName` value is the selector criteria for which resource to monitor. 
+
+The `dimensionValue` value is the value of the `dimensionName` you want to match.
+
+The `metricName` value is the metric you want to measure, these are different between namespaces.
+
+The `targetMetricValue` value is the average value you want to target.
+
+## Example
+
+[`examples/awscloudwatch_scaledobject.yaml`](./../../examples/awscloudwatch_scaledobject.yaml)
+
+## Using TriggerAuthentication
+
+This specification describes the `aws-role` TriggerAuthentication for AWS Cloudwatch.
+
+```yaml
```

Comment:
Nitpick - Let's add a `###` here with 'Role-based Access' or so and do same for SP auth for more structure in the spec?